src/statement-list/statement_list_entry_point.cpp

```diff
@@ -0,0 +1,179 @@
+/*******************************************************************\
+
+Module: Statement List Language Entry Point
+
+Author: Matthias Weiss, matthias.weiss@diffblue.com
+
+\*******************************************************************/
+
+/// \file
+/// Statement List Language Entry Point
+
+#include "statement_list_entry_point.h"
+#include "statement_list_typecheck.h"
+
+#include <goto-programs/goto_functions.h>
+#include <util/config.h>
+
+#define INITIALIZE_FUNCTION CPROVER_PREFIX "initialize"
+#define DB_ENTRY_POINT_POSTFIX "_entry_point"
+#define CPROVER_HIDE CPROVER_PREFIX "HIDE"
+
+bool statement_list_entry_point(
+  symbol_tablet &symbol_table,
+  message_handlert &message_handler,
+  const object_factory_parameterst &object_factory_parameters)
+{
+  // Check if the entry point is already present and return if it is.
+  if(
+    symbol_table.symbols.find(goto_functionst::entry_point()) !=
+    symbol_table.symbols.end())
+    return false;
+
+  irep_idt main_symbol;
+
+  // Find main symbol, if any is given.
+  if(config.main.has_value())
+  {
+    std::list<irep_idt> matches;
+
+    forall_symbol_base_map(
+      it, symbol_table.symbol_base_map, config.main.value())
+    {
+      symbol_tablet::symbolst::const_iterator s_it =
+        symbol_table.symbols.find(it->second);
+
+      if(s_it == symbol_table.symbols.end())
+        continue;
+
+      if(s_it->second.type.id() == ID_code)
+        matches.push_back(it->second);
+    }
+
+    if(matches.empty())
+    {
+      messaget message(message_handler);
+      message.error() << "main symbol `" << config.main.value() << "' not found"
+                      << messaget::eom;
+      return true;
+    }
+
+    if(matches.size() > 1)
+    {
+      messaget message(message_handler);
+      message.error() << "main symbol `" << config.main.value()
+                      << "' is ambiguous" << messaget::eom;
+      return true;
+    }
+
+    main_symbol = matches.front();
+  }
+  else
+  {
+    // TODO: Support the standard entry point of STL (organisation blocks).
+    // This also requires to expand the grammar and typecheck.
+    return false;
```

Comment:
In theory yes, returning `true` however prevents the symbol table from being generated (and shown via `--show-symbol-table`) if no `--function` is specified. For now it seems fine to return `false` since this gets caught later in the process and at the same time allows for checking the symbol table even without a proper entry point.